### ISSUE TITLE
generate_data_from_senat_url.sh never ends

### ISSUE THREAD

**Issue by promethe42** [regardscitoyens/the-law-factory-parser#47]
Hello,

I'm trying to generate data using generate_data_from_senat_url.sh:

`bash generate_data_from_senat_url.sh http://www.senat.fr/dossier-legislatif/ppl12-118.html`

I see a `perl` process taking 100% CPU but it never ends and nothing (except and empty data/ppl12-118/.tmp/dossier.csv` file) is created.

Sometimes, it works but it takes more than 5 minutes. Most of that time is spent being idle. I suspect it's waiting for some `curl` call to finish. Why would the requests take that much time?

Any idea?

**Comment by teymour** [comment]
The generators were developed to describe the evolution of bills during the parliamentary debate but the bill you want to load has never been discussed as it is in the state "Caducité".

edab222 corrects your problem allowing to load even these kinds of bills.

**Comment by RouxRC** [comment]
Hi @promethe42, Quitterie told me you were trying to use our parsers to collect data on bills for the legislative following process of MaVoix.
For such objective I believe most of what generate_data_from_senat_url does is not really useful for your use.
The whole data process chain is currently quite complex and messy making it hard to follow and fully get from an external viewpoint. We are currently considering refactoring a lot of it and make things more modular so that they would be more easily updatable for a live version of thelawfactory, or could be easily reusable for other projects. @fmassot started some first work on this in the parser-v2 branch
Let us know if you'd be interested in participating to this effort. Exchanging on our IRC #regardscitoyens@irc.frenode.net could be simplier then :)

**Comment by RouxRC** [comment]
closing this obsolete issue
